File: Assets/Scripts/PlayerMovement.cs
using UnityEngine;

public class PlayerMovement : MonoBehaviour
{

    [SerializeField] private float moveSpeed = 5f; // Speed of the player movement
    [SerializeField] private CharacterController controller;
    [SerializeField] private float turnSmoothTime;
    private float turnSmoothVelocity;

    private void Update()
    {
        //Vector2 direction = FixedJoystick.Instance.Direction;
        float horizontal = Input.GetAxis("Horizontal");
        float vertical = Input.GetAxis("Vertical");
        Vector3 direction = new Vector3 (horizontal,0f,vertical).normalized;

        if(direction.magnitude >= 0.1f)
        {
            float targetAngle = Mathf.Atan2(direction.x, direction.z) * Mathf.Rad2Deg /** Camera.main.transform.eulerAngles.y*/;
            float angle = Mathf.SmoothDampAngle(transform.eulerAngles.y, targetAngle, ref turnSmoothVelocity, turnSmoothTime);
            transform.rotation = Quaternion.Euler(0f, angle, 0f);

            //Vector3 moveDirection = Quaternion.Euler(0f, targetAngle, 0f) * Vector3.forward;
            controller.Move(direction.normalized * moveSpeed * Time.deltaTime);
        }
    }
}

File: Assets/Scripts/PlayerMovement.lua
-- local PlayerMovement = {}

-- function PlayerMovement.new()
--     local self = {}

--     self.moveSpeed = 5.0 -- Speed of the player movement
--     self.turnSmoothTime = 0.1
--     self.turnSmoothVelocity = 0.0
--     self.controller = nil

--     function self.Update()
--         local horizontal = Input.GetAxis("Horizontal")
--         local vertical = Input.GetAxis("Vertical")
--         local direction = Vector3.new(horizontal, 0.0, vertical).unit

--         if direction.magnitude >= 0.1 then
--             local targetAngle = math.atan2(direction.x, direction.z) * 180 / math.pi
--             local angle = math.smoothDampAngle(transform.eulerAngles.y, targetAngle, self.turnSmoothVelocity, self.turnSmoothTime)
--             transform.rotation = Quaternion.Euler(Vector3.new(0.0, angle, 0.0))

--             controller.Move(direction.unit * self.moveSpeed * Time.deltaTime)
--         end
--     end

--     return self
-- end

-- return PlayerMovement




function PrintThis( string)
    print(string)
end
